### PR TITLE
feat: add workspace instructions support (--workspace-instructions)

### PR DESCRIPTION
**Pull request by brrusino**
Add CLI flags and engine support for auto-discovering and injecting workspace instruction files (AGENTS.md, CLAUDE.md, .github/copilot-instructions.md) into all agent prompts.

- New instructions.py module: discovers convention files by walking from CWD up to git root, loads and concatenates them into a preamble
- CLI: --workspace-instructions (auto-discovery) and --instructions (explicit paths) flags on the run command
- Schema: instructions list field on WorkflowDef for YAML-level config
- Engine: preamble prepended to every agent's rendered prompt, inherited by sub-workflows, persisted in checkpoints for resume
- 32 new tests covering discovery, loading, injection, and edge cases

Closes microsoft/conductor#34

### CODE REVIEW

File: src/conductor/engine/workflow.py [hunk 1]
```diff
@@ -632,6 +648,17 @@ async def _execute_subworkflow(
...
+        # Merge instructions preamble: parent preamble + sub-workflow's own instructions
+        child_preamble = self._instructions_preamble
+        if sub_config.workflow.instructions:
+            from conductor.config.instructions import build_instructions_preamble
+
+            sub_preamble = build_instructions_preamble(
+                yaml_instructions=sub_config.workflow.instructions,
+            )
+            if sub_preamble:
+                child_preamble = child_preamble + sub_preamble if child_preamble else sub_preamble
```
Comment (jrob5756) [inline src/conductor/engine/workflow.py]: **Nested `<workspace_instructions>` tags when both parent and sub-workflow have preambles.**

`build_instructions_preamble()` always wraps its output in `<workspace_instructions>...</workspace_instructions>` tags. When the parent already has a preamble (also wrapped), this concatenation produces two complete blocks back-to-back:

```
<workspace_instructions>
parent stuff
</workspace_instructions>

<workspace_instructions>
sub stuff
</workspace_instructions>
```

This is semantically inconsistent with the single-block design used everywhere else and may confuse models that key on the wrapper boundaries.

Suggested fix: extract a helper that returns the inner (unwrapped) text and merge those, wrapping once at the outermost layer. For example, expose `_build_inner_preamble()` returning the raw concatenated content, and have `build_instructions_preamble()` wrap its result. The merge here would then combine the two inner texts under a single wrapper.

**Review by jrob5756 — COMMENTED** [review]
Really cool. Thanks for adding this. One small correctness fix.

File: src/conductor/engine/workflow.py (hunk 1 above)
Comment (brrusino) [reply src/conductor/engine/workflow.py]: Good catch — fixed in 8583be2. Refactored the preamble building into three helpers:

 - build_inner_instructions() — returns the raw concatenated content (no tags)
 - _wrap_preamble() — adds the <workspace_instructions> wrapper
 - _unwrap_preamble() — strips the wrapper for merging

The sub-workflow merge now unwraps the parent, concatenates inner content with the sub-workflow's instructions, and re-wraps once. Added a test that asserts exactly 1 opening and 1 closing tag in the merged output.

**Review by jrob5756 — APPROVED** [review]
LGTM
